test/unit/config.spec.js

```diff
@@ -5,6 +5,24 @@ const loadFile = require('mocks').loadFile
 const helper = require('../../lib/helper')
 const logger = require('../../lib/logger.js')
 
+/**
+ * If an object is "thenable", then it is considered to be a promise
+ * implementation. It does not have to be an instance of ECMAScript's own
+ * `Promise` class to be considered a promise.
+ *
+ * @param {*} value
+ * @returns {boolean}
+ *
+ * @see {@link https://promisesaplus.com/}
+ */
+function isPromiseLike (value) {
```

Comment:
Good point. parseConfig needs this logic because we don't know what kind of promise like object will be returned from the config file. Internally, Karma uses normal promises, so instanceOf will absolutely work for testing. I'll update this with the next push.

Reply:
Keep the second [message] argument? Since a helper function isn't being used, the matcher's default message may be enough.

```js

        expect(noConfigFilePromise).to.be.an.instanceof(
          Promise,
          'Expected parseConfig to return a promise when no config file path is provided.'
        )
        expect(syncConfigPromise).to.be.an.instanceof(
          Promise,
          'Expected parseConfig to return a promise when the config file DOES NOT return a promise.'
        )
        expect(asyncConfigPromise).to.be.an.instanceof(
          Promise,
          'Expected parseConfig to return a promise when the config file returns a promise.'
        )
```

Reply:
I made this initial change, but it appears to be failing. Even though a promise object is returned, both the matcher and the manual comparison of `noConfigFilePromise instanceof Promise` fails. I think I ran into something similar to this when working with jest. I could be misremembering, but it had to do with with a virtual machine or some other form of isolating the module under test. In this case Promise used by the module is different constructor than the Promise used by the test spec.

This is probably a side effect of `const loadFile = require('mocks').loadFile`

Reply:
We could work around it with: 

```js
expect(Object.getPrototypeOf(noConfigFilePromise).constructor.name).to.eql('Prototype')
```

But I think this is a poor practice, as it is vulnerable to any class, function, method, or object property with the name "Promise". It isn't necessarily ECMAScript's Promise implementation.

Reply:
Looks good!

Reply:
> Keep the second [message] argument? Since a helper function isn't being used, the matcher's default message may be enough.

Let's keep them.